datafusion/core/src/datasource/mod.rs

```diff
@@ -58,29 +57,33 @@ fn create_ordering(
     let mut all_sort_orders = vec![];
 
     for exprs in sort_order {
-        // Construct PhsyicalSortExpr objects from Expr objects:
-        let sort_exprs = exprs
-            .iter()
-            .map(|expr| {
-                if let Expr::Sort(Sort { expr, asc, nulls_first }) = expr {
-                    if let Expr::Column(col) = expr.as_ref() {
-                        let expr = expressions::col(&col.name, schema)?;
-                        Ok(PhysicalSortExpr {
-                            expr,
-                            options: SortOptions {
-                                descending: !asc,
-                                nulls_first: *nulls_first,
-                            },
-                        })
-                    } else {
-                        plan_err!("Expected single column references in output_ordering, got {expr}")
+        // Construct PhysicalSortExpr objects from Expr objects:
+        let mut sort_exprs = vec![];
+        for expr in exprs {
+            match expr {
+                Expr::Sort(sort) => match sort.expr.as_ref() {
+                    Expr::Column(col) => match expressions::col(&col.name, schema) {
+                        Ok(expr) => {
+                            sort_exprs.push(PhysicalSortExpr {
+                                expr,
+                                options: SortOptions {
+                                    descending: !sort.asc,
+                                    nulls_first: sort.nulls_first,
+                                },
+                            });
+                        }
+                        // Cannot find expression in the projected_schema, stop iterating
```

Comment:
This is necessary as the sort key may reference columns not found in the projected schema.

This change matches get_projected_output_ordering